datumaro/components/launcher.py

```diff
@@ -36,12 +38,20 @@ def __init__(self, extractor, launcher, batch_size=1):
 
     def __iter__(self):
         for batch in take_by(self._extractor, self._batch_size):
-            inputs = np.array([np.atleast_3d(item.media.data) for item in batch])
+            inputs = []
+            for item in batch:
+                if not isinstance(item.media, Image):
+                    raise MediaTypeError(
+                        f"Media type should be Image, Current type={type(item.media)}"
+                    )
+                inputs.append(np.atleast_3d(item.media.data))
+            inputs = np.array(inputs)
             inference = self._launcher.launch(inputs)
 
             for item, annotations in zip(batch, inference):
                 self._check_annotations(annotations)
-                yield self.wrap_item(item, annotations=annotations)
+                inferred_annotations = item.annotations + annotations
+                yield self.wrap_item(item, annotations=inferred_annotations)
```

Comment:
This change can also affect the others. Would you make a flag, e.g. `self.append (default=False)` and escalate it to `Dataset.run_model(..., append)`? Then, `Searcher` should call `dataset.run_model(..., append=True)`.

Reply:
Fix. I added `append_annotation` as instance variable which is only True for searcher.